### PR TITLE
add nginx.kak for highlighting detection filetype

### PR DESCRIPTION
**Pull request by nonumeros**

### CODE REVIEW

File: rc/filetype/nginx.kak [hunk 1]
```diff
@@ -0,0 +1,30 @@
+hook global BufCreate .*\.(nginx) %{
+    set-option buffer filetype nginx
```
Comment (krobelus) [inline rc/filetype/nginx.kak:2]: I don't think `*.nginx` files are a thing

Comment (nonumeros) [reply rc/filetype/nginx.kak:2]: I couldn't think of anything else here. You'e right. I'll add `params`, remove `nginx` and replace `*` by `+`. I guess that'll do it for now. I guess.

Comment (nonumeros) [reply rc/filetype/nginx.kak:2]: @krobelus  if you were to tell me now that say, the fastcgi_params and uwscgi_params are considered to be configuration files and as such such be included on the `conf.kak` instead, I would disagree. I mean. What would be the point of having those *files* there in the first place?

Comment (nonumeros) [reply rc/filetype/nginx.kak:2]: I'll go over it again @krobelus. On a system with gcc 11.2 works fine, whereas on 12.2 doesn't.

File: rc/filetype/nginx.kak [hunk 2]
```diff
@@ -0,0 +1,30 @@
+hook global BufCreate .*\(_params) %{
```
Comment (krobelus) [inline rc/filetype/nginx.kak:1]: Thanks but overall I'm not convinced this is worth merging.
It does not add anything, except on systems where `file(1)` is broken.
If there is a good reason for this change, please note it down.

Also you copied over parts of `conf.kak` that are definitely not valid in nginx files.

Comment (nonumeros) [reply rc/filetype/nginx.kak:1]: > Also you copied over parts of `conf.kak` that are definitely not valid in nginx files.

Yes. Perhaps some of it or most of it, is not valid for nginx files, but for the most part,  the few tidbits from it,  matched the syntax needed for nginx files accordingly.  Namely, as outlined on https://github.com/mawww/kakoune/pull/4765 which had as a premise that  "anything after and including *, except # where applicable,, is not commented out" still holds 

> Thanks but overall I'm not convinced this is worth merging. It does not add anything, except on systems where `file(1)` is broken. If there is a good reason for this change, please note it down.
> 

no problem. I don't know what you're referring to here with the `file(1)` part. 

If anything it's clear, kak on some systems, decides to write a file as an `x-c` while on others, it decides it's an `application/octream`. I wouldn't be surprised  if the preprocessor is to be  blamed for this erratic behavior, but that's besides the point, really.

Comment (nonumeros) [reply rc/filetype/nginx.kak:1]: This is the related https://github.com/mawww/kakoune/issues/4764

Comment (nonumeros) [reply rc/filetype/nginx.kak:1]: And this is the prior pull request that resolved the problem. 
https://github.com/mawww/kakoune/pull/4765

```
hook global BufCreate .+\.(conf|repo|cfg|properties|desktop) %{
```

Obviously that was from before the other revisions.

Comment (nonumeros) [reply rc/filetype/nginx.kak:1]: That's not even touching the fact that `hook global BufCreate .*` or `BufCreate .+`  is just for aesthetics really

Let  `nginx.conf` be  a new file  and `:write` it accordingly with kak. What do we have?

`$ xxd nginx.conf | head -1`

results:

`00000000: 0a                                       .`


Now, let emacs or neovim `write` a new `nginx.conf` file , or different `nginx.conf` file

`xxd` returns what is to be expected

Comment (nonumeros) [reply rc/filetype/nginx.kak:1]: > Thanks but overall I'm not convinced this is worth merging.

It is not. It doesn't need to be convincing :) But I've never seen a more contradictory statement for refusing to merge not only this one on its own, but also this one https://github.com/mawww/kakoune/pull/4765  from earlier that just had the addition of `conf` to 

```
hook global BufCreate .+\.(conf|repo|cfg|properties|desktop) %{
```

In the meantime xxd shows anything written with kak as a

```
00000000: 0a                                       .
```

but with the same token, it `can't ` detect syntax highlighting accordingly for any new `nginx.conf` file. That's beyond me.

Comment (krobelus) [reply rc/filetype/nginx.kak:1]: the 0a you are seeing in the xxd output is https://github.com/mawww/kakoune/issues/2147, there are usable workarounds and proposed fixes

**Pull request closed without merging**